test/controllers/user.controller.test.js

```diff
@@ -3184,12 +3192,19 @@ describe('User Controller', () => {
         }),
       );
     });
+
     context('when multiple filters are used', () => {
+      const filteredUrl = JSON.stringify(dummyVendor1Details)
```

Comment:
I think it is better to use the queryString to generate the filter params. Also, the name should be filteredParams as it contains a list of parameters not the entire URL.

```suggestion
      const filteredUrl = querystring.stringify(dummyVendor1Details);
```

You need to import querystring at the top of the test file to use the above

```js
import querystring from 'querystring';
```

Please take a minute to go through it in the documentation since it contains lots of valuable information to cater for query strings.

https://nodejs.org/api/querystring.html